src/core/ngx_connection.c

```diff
@@ -487,6 +488,13 @@ ngx_open_listening_sockets(ngx_cycle_t *cycle)
                 continue;
             }
 
+#if (NGX_LINUX)
```

Comment:
if we can can do that on other OS, then you don't need this check here, just making sure `protocol` is initialised to 0 (I think it is)

(e.g. `lsopt` used below has been explicitly `ngx_memzero()`)

Reply:
in the same block, would it be considered to expensive to always call socket twice if it failed the first time?

Reply:
The idea would be that the protocol is > 0 only in Linux.

Reply:
ok, done then